netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -255,7 +242,13 @@ private NettyClientHandler newHandler() {
     frameWriter = new Http2OutboundFrameLogger(frameWriter, frameLogger);
 
     BufferingHttp2ConnectionEncoder encoder = new BufferingHttp2ConnectionEncoder(
-            new DefaultHttp2ConnectionEncoder(connection, frameWriter));
+            new DefaultHttp2ConnectionEncoder(connection, frameWriter)) {
+      @Override
+      public ChannelFuture writeSettingsAck(ChannelHandlerContext ctx, ChannelPromise promise) {
+        listener.transportReady();
```

Comment:
That assumes the remote implementation does not do something it is entitled to do, which is inappropriate. The server is fully entitled to send SETTINGS frames at any time.

At the moment [we do actually send an extra SETTINGS frame](https://github.com/grpc/grpc-java/blob/master/netty/src/main/java/io/grpc/netty/NettyServerHandler.java#L365) when connecting. That's because Netty sends one internally but didn't let us modify it before it was sent, so we had to send our own. That'll be resolved in the next Netty beta, though.